go/host/l1/publisher.go

```diff
@@ -145,32 +161,29 @@ func (p *Publisher) PublishSecretResponse(secretResponse *common.ProducedSecretR
 	return nil
 }
 
-func (p *Publisher) ExtractSecretResponses(block *types.Block) []*ethadapter.L1RespondSecretTx {
+func (p *Publisher) ExtractObscuroRelevantTransactions(block *types.Block) ([]*ethadapter.L1RespondSecretTx, []*ethadapter.L1RollupTx, []*ethadapter.L1SetImportantContractsTx) {
 	var secretRespTxs []*ethadapter.L1RespondSecretTx
+	var rollupTxs []*ethadapter.L1RollupTx
+	var contractAddressTxs []*ethadapter.L1SetImportantContractsTx
```

Comment:
something for another PR, maybe , but we need to start using events here instead of rummaging through transactions.
Let's add a ticket/todo

Reply:
yeah good shout, I'll add a todo comment and ticket